File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        int number=0;
        String strX = Integer.toString(x);
        
        for(int i=0; i<strX.length(); i++){
            number += Character.getNumericValue(strX.charAt(i));
        }
        
        if(x%number!=0){
            answer=false;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.js
function solution(x) {
    var strX = x.toString();
    var num = 0;
    
    for(var i=0; i<strX.length; i++){
        num += Number(strX.charAt(i));    
    }
    
    if(x%num == 0){
        return true;
    } else {
        return false;
    }
}